malariagen_data/af1.py

```diff
@@ -0,0 +1,747 @@
+            right="right",
+            color="#cccccc",
+            source=df_n_runs,
+            name="gaps",
+        )
+
+        debug("plot SNPs")
+        color_pass = bkpal.Colorblind6[3]
+        color_fail = bkpal.Colorblind6[5]
+        data["left"] = data["pos"] - 0.4
+        data["right"] = data["pos"] + 0.4
+        data["bottom"] = np.where(data["is_seg"], 1.6, 0.6)
+        data["top"] = data["bottom"] + 0.8
+        data["color"] = np.where(data[f"pass_{site_mask}"], color_pass, color_fail)
+        fig.quad(
+            top="top",
+            bottom="bottom",
+            left="left",
+            right="right",
+            color="color",
+            source=data,
+            name="snps",
+        )
+        # TODO add legend?
+
+        debug("tidy plot")
+        fig.yaxis.ticker = bkmod.FixedTicker(
+            ticks=[1, 2],
+        )
+        fig.yaxis.major_label_overrides = {
+            1: "Non-segregating",
+            2: "Segregating",
+        }
+        fig.xaxis.axis_label = f"Contig {region.contig} position (bp)"
+        fig.xaxis.ticker = bkmod.AdaptiveTicker(min_interval=1)
+        fig.xaxis.minor_tick_line_color = None
+        fig.xaxis[0].formatter = bkmod.NumeralTickFormatter(format="0,0")
+
+        if show:
+            bkplt.show(fig)
+
+        return fig
+
+    def plot_snps(
+        self,
+        region,
+        sample_sets=None,
+        sample_query=None,
+        site_mask=DEFAULT_SITE_MASK,
+        width=800,
+        track_height=80,
+        genes_height=120,
+        show=True,
+    ):
+        # TODO docstring
+        debug = self._log.debug
+
+        import bokeh.layouts as bklay
+        import bokeh.plotting as bkplt
+
+        debug("plot SNPs track")
+        fig1 = self.plot_snps_track(
+            region=region,
+            sample_sets=sample_sets,
+            sample_query=sample_query,
+            site_mask=site_mask,
+            width=width,
+            height=track_height,
+            show=False,
+        )
+        fig1.xaxis.visible = False
+
+        debug("plot genes track")
+        fig2 = self.plot_genes(
+            region=region,
+            width=width,
+            height=genes_height,
+            x_range=fig1.x_range,
+            show=False,
+        )
+
+        fig = bklay.gridplot(
+            [fig1, fig2], ncols=1, toolbar_location="above", merge_tools=True
+        )
+
+        if show:
+            bkplt.show(fig)
+
+        return fig
+
+    @staticmethod
+    def _setup_taxon_colors(plot_kwargs):
+        import plotly.express as px
+
+        taxon_palette = px.colors.qualitative.Plotly
+        taxon_color_map = {
+            "funestus": taxon_palette[0],
+            "gcx1": taxon_palette[1],
+            "gcx2": taxon_palette[2],
+            "gcx3": taxon_palette[3],
+        }
+        plot_kwargs["color_discrete_map"] = taxon_color_map
+        plot_kwargs["category_orders"] = {"taxon": list(taxon_color_map.keys())}
```

Comment:
The taxon values "gcx1", "gcx2" and "gcx3" are only applicable to Ag3. 
```suggestion
    @staticmethod
    def _setup_taxon_colors(plot_kwargs):
        import plotly.express as px

        taxon_palette = px.colors.qualitative.Plotly
        taxon_color_map = {
            "funestus": taxon_palette[0],
        }
        plot_kwargs["color_discrete_map"] = taxon_color_map
        plot_kwargs["category_orders"] = {"taxon": list(taxon_color_map.keys())}
```

Reply:
Of course, thanks! Will fix.